kite-data/kite-data-crunch/src/main/java/org/kitesdk/data/crunch/DatasetTarget.java

```diff
@@ -85,7 +85,14 @@ public boolean handleExisting(WriteMode writeMode, long lastModForSource,
       }
     }
 
-    boolean exists = !view.isEmpty();
+    boolean ready = false;
+    try {
+      ready = view.isReady();
+    } catch (UnsupportedOperationException ex) {
+      // view isn't readiable, continue as if it wasn't ready
+    }
+    // a view exists if it isn't empty, or if it has been marked ready
+    boolean exists = !view.isEmpty() || ready;
```

Comment:
This change makes a couple behavior updates. 

In WriteMode.DEFAULT a view could be considered to exist if it was empty but had been signed as ready. So in normal operation if a crunch stage (successfully) had an output view that received no data, but was marked as "ready", a subsequent run would fail with the RuntimeException in the case statement below. Before this would only have happened if the dataset/view was not empty.

In the CHECKPOINT case the same changes around what constitutes a View that exists also apply. Beyond that in checkpoint mode the pipeline would be fast-forwarded to that output point if the output view is newer than the input, and the output is signaled ready. Otherwise if the output view is overwritten if the source view has been modified more recently, or if the output view was never signaled ready. For example if an external process wrote to a view, but didn't signal it, checkpoint would remove the existing data and produce new output data.

It's worth noting too that the PR updates the "last modified" concept to consider the time a view was marked as ready. So if I produced a starting dataset at T0, validated it, and then marked it ready at T1, the view would be considered as modified at T1. (if it was never marked as ready, it will still evaluate last modified as T0)